### DISCUSSION TITLE
background image behavior

### DISCUSSION THREAD

**Discussion by DanielGennat** [neuefische/questions#160] [Web] [answered]
Hallo zusammen,

wie kann ich in nextJS ein Hintergrundbild einfügen, dass sich folgendermaßen verhält:
- die Höhe des Bildes soll den Viewport ausfüllen
- die Bildmitte soll an der Mitte des Viewports ausgerichtet sein, sodass nur die linke und rechte Seite des Bildes verschwinden, wenn der viewport zu klein ist (das betrifft vor allem die mobile portrait Ansicht)
- als next/image (falls das möglich ist, ich finde per Googlesuche die entsprechenden props nicht heraus)

Das Bild ist ursprünglich ein .jpg. Aktuell verwende ich es als .png (export per Figma).

Hier noch eine Skizze, um das gewünschte Ergebnis zu veranschaulichen:
![background image behavior](https://user-images.githubusercontent.com/109812125/191754389-74da908f-8f26-4168-9415-09bd1704d324.png)

Gefundene Ansätze:
- [mit className den z-index verändern](https://uharston.medium.com/next-js-image-optimization-on-background-images-65de18ea03f5), aber dann bräuchte ich eine .css Datei, was dann zu einem Mix mit styled components führen würde
- in der [mdn Dokumentation](https://nextjs.org/docs/api-reference/next/image) wird ein [background Beispiel](https://image-component.nextjs.gallery/background) aufgeführt. Das ist das Verhalten, das ich haben will, aber `background` wird nicht als Prop gelistet und funktioniert auch nicht als Prop.

Kann ich das Bild also nur über einen styled `<img>` mit einem styled `span` einfügen, so wie in den DevTools beim background Beispiel, oder gibt es da noch eine schönere Lösung mit next/image?

[Link zum Pullrequest](https://github.com/DanielGennat/capstone-project/pull/8)

Vielen Dank im Voraus :)

**Comment by NikRoe** [comment, marked as answer]
Hey Daniel,
can you provide us with a link to the PR?

Cheers :)

**Reply by DanielGennat** [reply]
Hey Niklas, link is added :)

**Reply by DanielGennat** [reply]
mir ist grad aufgefallen, dass ich das Bild in der Mitte ausrichten möchte. Ich update die Frage nochmal. Die Ausrichtung am Rand interessiert mich aber trotzdem.

**Reply by DanielGennat** [reply]
updated

**Reply by DanielGennat** [reply]
sorry für das viele Updaten, aber jetzt müsste die Frage vollständig sein :)

**Reply by NikRoe** [reply]
All good, all good :)

the example provided by NextJS is actually a really nice starting point and luckily we don't need to use CSS modules but can handle everything with `styled-components`.

You can wrap your `next/image` component in a container and style that container. It could look like this:
```js
<ImageWrapper>
        <Image
          alt="background image man playing guitar on his back"
          src={manPlayingGuitarOnHisBackImg}
          layout="fill"
          objectFit="cover"
        />
</ImageWrapper>
```

The corresponding container could look like this:

```js
const ImageWrapper = styled.div`
  position: fixed;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
  z-index: -1;
`;
```

On another note: You should implement the already created `GlobalStyles.js` in your `_app.js` ;)

Hope this already works, let me know otherwise :)

**Reply by DanielGennat** [reply]
Vielen Dank, jetzt funktioniert es!
`objectFit="cover"` ist die Zeile die gefehlt hat.
Den wrap brauche ich dann wahrscheinlich nur, wenn andere Elemente nicht über dem Bild angezeigt werden, aber bislang ist es so wie es sein soll :)
